dali/pipeline/graph_descr.cc

```diff
@@ -71,358 +73,444 @@ void CheckOpConstraints(const OpSpec &spec) {
       + " outputs, but was passed " + std::to_string(spec.NumOutput()/num_input_sets) + ".");
 }
 
+
+DALIOpType DeviceStringToOpType(std::string device) {
```

Comment:
`ParseOpType`? Also, passing string by value is a bad idea if you're not going to store it.

Reply:
Done.

Reply:
Where?

Reply:
Sorry, turns out I didn't push the new one to proper branch.